core/error_api.php

```diff
@@ -480,30 +488,47 @@ function error_print_context( array $p_context ) {
 }
 
 /**
- * Print out a stack trace
+ * Get the stack trace as a string that can be logged or echoed to CLI output.
+ *
+ * @param Exception|null $p_exception The exception to print stack trace for.  Null will check last seen exception.
+ * @return string multi-line printout of stack trace.
  */
-function error_print_stack_trace() {
-	$t_stack = error_stack_trace();
+function error_stack_track_as_string( $p_exception = null ) {
+	$t_stack = error_stack_trace( $p_exception );
+	$t_output = '';
 
-	if( php_sapi_name() == 'cli' ) {
-		foreach( $t_stack as $t_frame ) {
-			echo ( isset( $t_frame['file'] ) ? $t_frame['file'] : '-' ), ': ' ,
-				( isset( $t_frame['line'] ) ? $t_frame['line'] : '-' ), ': ',
-				( isset( $t_frame['class'] ) ? $t_frame['class'] : '-' ), ' - ',
-				( isset( $t_frame['type'] ) ? $t_frame['type'] : '-' ), ' - ',
-				( isset( $t_frame['function'] ) ? $t_frame['function'] : '-' );
-	
-			$t_args = array();
-			if( isset( $t_frame['args'] ) && !empty( $t_frame['args'] ) ) {
-				foreach( $t_frame['args'] as $t_value ) {
-					$t_args[] = error_build_parameter_string( $t_value );
-				}
-				echo '(', implode( $t_args, ', ' ), ' )', "\n";
-			} else {
-				echo "()\n";
+	foreach( $t_stack as $t_frame ) {
+		$t_output .= ( isset( $t_frame['file'] ) ? $t_frame['file'] : '-' ) . ': ' .
+			( isset( $t_frame['line'] ) ? $t_frame['line'] : '-' ) . ': ' .
+			( isset( $t_frame['class'] ) ? $t_frame['class'] : '-' ) . ' - ' .
+			( isset( $t_frame['type'] ) ? $t_frame['type'] : '-' ) . ' - ' .
+			( isset( $t_frame['function'] ) ? $t_frame['function'] : '-' );
+
+		$t_args = array();
+		if( isset( $t_frame['args'] ) && !empty( $t_frame['args'] ) ) {
+			foreach( $t_frame['args'] as $t_value ) {
+				$t_args[] = error_build_parameter_string( $t_value );
 			}
+
+			$t_output .= '( ' . implode( $t_args, ', ' ) . " )\n";
+		} else {
+			$t_output .= "()\n";
 		}
-	
+	}
+
+	return $t_output;
+}
+
+/**
+ * Print out a stack trace
+ *
+ * @param Exception|null $p_exception The exception to print stack trace for.  Null will check last seen exception.
+ */
+function error_print_stack_trace( $p_exception = null ) {
+	$t_stack = error_stack_trace( $p_exception );
```

Comment:
Move that after the _cli_ test, just before the foreach loop

Reply:
Moved.